src/components/tl/sharp/tl_sharp_coll.c

```diff
@@ -160,18 +160,39 @@ void ucc_tl_sharp_collective_progress(ucc_coll_task_t *coll_task)
     if (task->req_handle != NULL) {
         completed = sharp_coll_req_test(task->req_handle);
         if (completed) {
-            if (TASK_ARGS(task).coll_type == UCC_COLL_TYPE_ALLREDUCE) {
+            switch(TASK_ARGS(task).coll_type){
+            case UCC_COLL_TYPE_ALLREDUCE:
                 if (!UCC_IS_INPLACE(TASK_ARGS(task))) {
                     ucc_tl_sharp_mem_deregister(TASK_TEAM(task),
                                                 task->allreduce.s_mem_h);
                 }
                 ucc_tl_sharp_mem_deregister(TASK_TEAM(task),
                                             task->allreduce.r_mem_h);
-            }
-            if (TASK_ARGS(task).coll_type == UCC_COLL_TYPE_BCAST) {
+                break;
+            case UCC_COLL_TYPE_BCAST:
                 ucc_tl_sharp_mem_deregister(TASK_TEAM(task),
                                             task->bcast.mem_h);
+                break;
+            case UCC_COLL_TYPE_ALLGATHER:
+                if (!UCC_IS_INPLACE(TASK_ARGS(task))) {
+                    ucc_tl_sharp_mem_deregister(TASK_TEAM(task),
```

Comment:
we should check status, even though IIUC `ucc_tl_sharp_mem_deregister` cannot return any error -- in this case we might want to change the return type to void

Reply:
done